src/server/game/Movement/PathGenerator.cpp

```diff
@@ -482,8 +482,9 @@ void PathGenerator::BuildPointPath(const float *startPoint, const float *endPoin
         memcpy(&pathPoints[VERTEX_SIZE * 0], startPoint, sizeof(float)* 3); // first point
 
         // path has to be split into polygons with dist SMOOTH_PATH_STEP_SIZE between them
-        G3D::Vector3 startVec = G3D::Vector3(startPoint[0], startPoint[1], startPoint[2]);
-        G3D::Vector3 endVec = G3D::Vector3(endPoint[0], endPoint[1], endPoint[2]);
+		//startPoint and endPoint are in format y,z,x while G3D::Vector3 expects x,y,z
```

Comment:
BuildPointPath is called from BuildPolyPath which passes on 2 float pointers, namely : 
` float startPoint[VERTEX_SIZE] = {startPos.y, startPos.z, startPos.x};`
` float endPoint[VERTEX_SIZE] = {endPos.y, endPos.z, endPos.x};`

As you can see, they're not in the usual x, y, z format while they're being used that way when constructing 2 3D vectors in BuildPointPath, which is wrong.